core/src/main/scala/org/apache/spark/util/Clock.scala

```diff
@@ -21,12 +21,22 @@ package org.apache.spark.util
  * An interface to represent clocks, so that they can be mocked out in unit tests.
  */
 private[spark] trait Clock {
+
+  /**
+   * @return the time in milliseconds
+   */
   def getTimeMillis(): Long
+
+  /**
+   * @param targetTime block until the current time is at least this value
+   * @return current system time when wait has completed
```

Comment:
Not sure how that's related to my comment. I'm just saying that this method doesn't wait until "system time", but whatever is the clock's definition of time.